assets/js/phoenix_live_view/js.js

```diff
@@ -54,9 +54,15 @@ let JS = {
     })
   },
 
-  exec_dispatch(e, eventType, phxEvent, view, sourceEl, el, {event, detail, bubbles}){
+  exec_dispatch(e, eventType, phxEvent, view, sourceEl, el, {event, detail, bubbles, blocking}){
     detail = detail || {}
     detail.dispatcher = sourceEl
+    if(blocking){
+      const promise = new Promise((resolve, _reject) => {
+        detail.done = resolve
```

Comment:
Should we error if `detail.done` already exists? Seems like something that could already be in use. That being said, since the user has to opt-in to blocking, it's not going to break for anyone upgrading LV.

Reply:
I added a check in `Phoenix.LiveView.JS.dispatch/3`. More likely for people to see the error on the server than in the browser console.